### PR TITLE
Rewrite install task

### LINKED ISSUE #1909: Use Task class to rewrite some function-based tasks
- [ ] `virtool.software.db.install()`
- [ ] `virtool.hmm.db.install()`

### PR DESCRIPTION
**Pull request by tianshengsui**
resolve #1909

### CODE REVIEW

File: virtool/hmm/db.py [hunk 1]
```diff
@@ -38,6 +38,125 @@
 ]
 
 
+class HmmInstallTask(virtool.tasks.task.Task):
```
Comment (igboyes) [inline virtool/hmm/db.py:41]: Rename to `HMMInstallTask`.

**Review by igboyes — CHANGES_REQUESTED** [review]
One rename.

Why didn't you split the task functions into separate steps when moving to `Task`?

**Comment by tianshengsui** [comment]
> One rename.
> 
> Why didn't you split the task functions into separate steps when moving to `Task`?

Because all the code are under the context of ```tempdir```. If I split them into several functions, I think the ```tempdir``` will be closed and removed.

**Comment by igboyes** [comment]
Okay. That is what I thought.

Can you add the tempdir generation and cleanup to the base `Task` class and make `temp_dir` an attribute on that class? That way the temporary directory is available to all the tasks but is still automatically deleted at the end.

**Comment by tianshengsui** [comment]
I will do it.

**Review by igboyes — APPROVED** [review]

**Pull request merged**